hibernate-core/src/main/java/org/hibernate/id/ForeignGenerator.java

```diff
@@ -116,10 +115,12 @@ public Serializable generate(SharedSessionContractImplementor sessionImplementor
 			if (sessionImplementor instanceof Session) {
 				id = ((Session) sessionImplementor)
 						.save(foreignValueSourceType.getAssociatedEntityName(), associatedObject);
-			} else if (sessionImplementor instanceof StatelessSession) {
+			}
+			else if (sessionImplementor instanceof StatelessSession) {
```

Comment:
could you squash the git commits into the main one?

Reply:
I squashed and moved static import to the last. Still need a testing case.

Reply:
I created a testing case and squashed everything. Enjoy!

Reply:
@Sanne could you take a new look? I think it has been improved enormously after a well-thought testing case was added.